src/main/java/cn/nukkit/command/defaults/BanIpCommand.java

```diff
@@ -43,31 +46,29 @@ public boolean execute(CommandSender sender, String commandLabel, String[] args)
         }
 
         if (args.length == 0) {
-            sender.sendMessage(new TranslationContainer("commands.generic.usage", this.usageMessage));
-
             return false;
         }
 
         String value = args[0];
-        String reason = "";
+        StringBuilder reason = new StringBuilder();
         for (int i = 1; i < args.length; i++) {
-            reason += args[i] + " ";
+            reason.append(args[i]).append(" ");
         }
 
         if (reason.length() > 0) {
-            reason = reason.substring(0, reason.length() - 1);
+            reason = new StringBuilder(reason.substring(0, reason.length() - 1));
```

Comment:
`reason.setLength(reason.length() - 1);`

no reason to allocate a whole new `StringBuilder` for nothing

Reply:
This was IntelliJ's smart fix, when it had a code analysis of string += in a loop, and quick fix was to change to StringBuilder. Good catch, I'll fix.